.storybook/main.js

```diff
@@ -59,10 +56,28 @@ module.exports = {
       },
       resolve: {
         ...config.resolve,
-        modules: custom.resolve.modules,  // important for absolute path resolution
+        modules: custom.resolve.modules, // important for absolute path resolution
         // enable resolve override sparingly
         // storybook config includes many resolve.alias and resolve.plugins
         // ...custom.resolve,
+        plugins: [
+          ...config.resolve.plugins,
+          // Custom alias plugin required for webpack 4
+          // See https://github.com/webpack/webpack/issues/6817#issuecomment-542448438
+          new AliasPlugin(
+            'described-resolve',
+            [
+              {
+                name: 'hooks',
+                alias: [
```

Comment:
mm, I still didn't understand why u want to override the hook?
I mean, overriding a hook wouldn't be a red flag for a bad or complex module architecture?

Reply:
Think of it as a different approach to the same problem addressed by the API mocks we have.

Besides testing only dumb components, we could also use Storybook stories for implementing complex components composed of smaller dumber components, like full pages.

One example: think about the swap app's Swap page.
It relies heavily on hooks, aggregates several other small components and have some styled sneaking in some parts https://github.com/gnosis/gp-swap-ui/blob/develop/src/pages/Swap/index.tsx#L306

In order to make any UI changes, you'd normally start the change the code, refresh, etc.
But, depending on what step in the UI you want to test, there are network calls, intermediate states, and so on.
For this I think Storybook is of great value.
You can mock only key hooks (most of them can remain unchanged, depending on what they do behind the scenes) to get to the state you want and be able to view/modify only the UI.

I agree with the redux approach of providing the data, but there are also external calls we might not want to do on Storybook.

The reason why I wanted to bring it to this project no longer applies because there's another solution, and I don't have a use case for that right now.
More example I can give you are from CMM where smaller components had hooks, like this one:

Stories definition: https://github.com/gnosis/safe-cmm-app-react/blob/development/src/components/basic/display/MarketPrice/MarketPrice.stories.tsx
Actual stories: https://safe-cmm.gnosis.io/storybook/?path=/story/basic-display-marketprice--default
Main component: https://github.com/gnosis/safe-cmm-app-react/blob/development/src/components/basic/display/MarketPrice/index.tsx

The 'view' doesn't have any hooks and don't need any mocking.
The 'container' does, and there are also stories for it to "mock" the component behaviour under different hook responses.

Since this is sorta of a basic component - meaning, won't have a page by itself - it'll be included in another story, and there it'll need to mock it's inner hooks.

Having said all that, I can't say I'm happy with all design decisions we made in the CMM app and that there are no technical debts to address. 

Does that make it any clearer?

Reply:
Thanks for providing this detailed context, yes it makes it clearer. Just for the effort to document your thought I'm approving the PR ✅

By the way, storybook has also ways to intercept things like clicks, but I would also try to avoid that.
Okay, we'll see when we start to compose the components and we can decide in a per case basis if there's alternatives. 
Thanks!